src/NBitcoin/ChainedBlock.cs

```diff
@@ -305,12 +355,6 @@ public DateTimeOffset GetMedianTimePast()
             return median[begin + ((end - begin) / 2)];
         }
 
-        private static void assert(object obj)
-        {
-            if (obj == null)
-                throw new NotSupportedException("Can only calculate work of a full chain");
-        }
-
         /// <summary>
         /// Check PoW and that the blocks connect correctly.
```

Comment:
we need to update the comment of ` public bool Validate(Network network) ` as it is not correct, there is PoS validation as well

Reply:
If I remember correctly POS validation happens in another place.

Reply:
this is not full validation, just headers validation with BlockStake.Validate